crates/cheatnet/src/rpc.rs

```diff
@@ -358,6 +359,65 @@ impl CheatableSyscallHandler<'_> {
         ptr_cheated_block_info
     }
 
+    fn get_cheated_tx_info_ptr(
+        &self,
+        vm: &mut VirtualMachine,
+        original_tx_info: &[MaybeRelocatable],
+        contract_address: &ContractAddress,
+    ) -> Relocatable {
+        // create a new segment with replaced tx info
+        let ptr_cheated_tx_info = vm.add_memory_segment();
+
+        let mut new_tx_info = original_tx_info.to_owned();
+
+        if let Some(tx_info_mock) = self.cheatcode_state.spoofed_contracts.get(contract_address) {
+            let TxInfoMock {
```

Comment:
Shouldn't we panic in case it's `None`? `get_cheated_tx_info_ptr` is only executed if contract was in fact spoofed, so I'm assuming it should be impossible for this function to execute with it being `None`.

Reply:
I think unwrapped lookup is enough.